pallets/deeper-node/src/lib.rs

```diff
@@ -58,16 +68,24 @@ decl_module! {
         fn deposit_event() = default;
 
         #[weight = 10_000]
-        pub fn register_device(origin, ip: Vec<u8>, country: u16) -> DispatchResult {
+        pub fn register_device(origin, ip: Vec<u8>, country: Vec<u8>) -> DispatchResult {
             let sender = ensure_signed(origin)?;
-            ensure!(ip.len() < 256, "invalid ip address");
-            ensure!(country < 1024, "invalid country code");
+
+            if RegionMapInit::get() == false {
+                let _ = Self::setup_region_map();
+                RegionMapInit::put(true);
+            }
+
+            ensure!(RegionMap::contains_key(&country), "invalid country code");
+            ensure!(ip.len() <= 256, "invalid ip address");
+
             let node = Node {
                 account_id: sender.clone(),
                 ipv4: ip.clone(),
-                country: country,
+                country: country.clone(),
+                expire: <frame_system::Module<T>>::block_number(),
```

Comment:
it seems that the expire equals the current block number when register_device is called.